### ISSUE TITLE
Stuck in full screen

### ISSUE THREAD

**Issue by DesertCookie** [jellyfin/jellyfin-media-player#94]
**Jellyfin Media Player is stuck in full screen after I exited it via ALT+F4 while watching a movie in full screen.**

I've tried restarting, uninstalling, manually disabling the `fullscreen` setting in the client options, starting a movie and going back to windowed mode that way, F11, ALT+ENTER.

This occurred on Windows 10 20H2 with JMP v1.6 using the Windows installer. See my [JellyfinMediaPlayer.log](https://github.com/jellyfin/jellyfin-media-player/files/6517510/JellyfinMediaPlayer.log).

---

**Solution**: Delete `jellyfinmediaplayer.conf` in `%APPDATA%/Local/JellyfinMediaPlayer`.

Please look into why this happened to prevent future users from having to manually delete the config file. Cheers.

**Comment by mekemahmud** [comment]
Search for "jellyfinmediaplayer.conf" file in C drive and open the file in notepad when you find the file. Find the "forceFSScreen": 
 in "main" section and change it's value from "DISPLAY2" to "DISPLAY1". I also was facing the same issue for last few days and just an hour ago figured out the problem and solved it.
Hope it helps.
thanks.
![Capture](https://user-images.githubusercontent.com/89276060/130286299-9c6098ce-69c6-43de-aed5-26b96e754dc0.JPG)

**Comment by foxlet** [comment]
Are there any fixes for this? It still occurs in the latest stable version (v1.6.1).

**Comment by davidmurdoch** [comment]
Not sure if I have the same problem since it's always Fullscreen for me, except for the Windows Taskbar at the bottom. Is there supposed to be be way of exiting the app (traditionally via an X in the upper right corner in Windows)? Alt+F4 or right clicking the taskbar entry and clicking Close Window seem like an odd thing.

**Comment by DexLuther** [comment]
I'm having the same issue.

Jellyfin Server 1.7.7
Jellyfin Media Player 1.7.0-Pre4

* F11
   * Briefly flashes into Windowed mode for s split second, then reverts back to  Fullscreen.
* ALT+Enter
   * Does nothing.
   * Spamming ALT+ENTER repeatedly flashes the screen. A Slight blur can be seen. Looks like the program is attempting to resize into windowed mode, but immediately snaps back to fullscreen.
* Client Settings
   * Fullscreen Checkbox
      * Does nothing when on or off.
   * Turning off Fullscreen, clicking save and then returning to the Client settings window.
      * The fullscreen check box is checked again.

The fullscreen checkbox seems to automatically revert to being checked (forceFSScreen:true)

Logs show this:

```
2022-06-03 03:02:00 [ DEBUG ] KonvergoWindow.cpp @ 479 - QWindow visibility set to QWindow::Windowed 
2022-06-03 03:02:00 \[ WARN  \] KonvergoWindow.cpp @ 487 - winging it! 
2022-06-03 03:02:00 \[ DEBUG \] KonvergoWindow.cpp @ 596 - resize event: QSize(1938, 1127) 
2022-06-03 03:02:00 \[ DEBUG \] KonvergoWindow.cpp @ 596 - resize event: QSize(1920, 1080) 
2022-06-03 03:02:00 \[ DEBUG \] KonvergoWindow.cpp @ 479 - QWindow visibility set to QWindow::FullScreen 
2022-06-03 03:02:00 \[ DEBUG \] KonvergoWindow.cpp @ 257 - Restoring geo: QRect(4110,129 982x551) 
2022-06-03 03:02:00 \[ DEBUG \] KonvergoWindow.cpp @ 209 - Load FullScreen geo... 
2022-06-03 03:02:17 \[ DEBUG \] KonvergoWindow.cpp @ 294 - setting fullscreen =  false 
2022-06-03 03:02:17 \[ DEBUG \] KonvergoWindow.cpp @ 479 - QWindow visibility set to QWindow::Windowed 
2022-06-03 03:02:17 \[ WARN  \] KonvergoWindow.cpp @ 487 - winging it! 
2022-06-03 03:02:17 \[ DEBUG \] KonvergoWindow.cpp @ 596 - resize event: QSize(1938, 1127) 
2022-06-03 03:02:17 \[ DEBUG \] KonvergoWindow.cpp @ 596 - resize event: QSize(1920, 1080) 
2022-06-03 03:02:17 \[ DEBUG \] KonvergoWindow.cpp @ 479 - QWindow visibility set to QWindow::FullScreen 
2022-06-03 03:02:17 \[ DEBUG \] KonvergoWindow.cpp @ 257 - Restoring geo: QRect(4110,129 982x551) 
2022-06-03 03:02:17 \[ DEBUG \] KonvergoWindow.cpp @ 209 - Load FullScreen geo... 
2022-06-03 03:02:18 \[ DEBUG \] KonvergoWindow.cpp @ 294 - setting fullscreen =  false 
2022-06-03 03:02:18 \[ DEBUG \] KonvergoWindow.cpp @ 479 - QWindow visibility set to QWindow::Windowed 
2022-06-03 03:02:18 \[ WARN  \] KonvergoWindow.cpp @ 487 - winging it! 
2022-06-03 03:02:18 \[ DEBUG \] KonvergoWindow.cpp @ 596 - resize event: QSize(1938, 1127) 
2022-06-03 03:02:18 \[ DEBUG \] KonvergoWindow.cpp @ 596 - resize event: QSize(1920, 1080) 
2022-06-03 03:02:18 \[ DEBUG \] KonvergoWindow.cpp @ 479 - QWindow visibility set to QWindow::FullScreen 
2022-06-03 03:02:18 \[ DEBUG \] KonvergoWindow.cpp @ 257 - Restoring geo: QRect(4110,129 982x551) 
2022-06-03 03:02:18 \[ DEBUG \] KonvergoWindow.cpp @ 209 - Load FullScreen geo... 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 294 - setting fullscreen =  false 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 479 - QWindow visibility set to QWindow::Windowed 
2022-06-03 03:02:20 \[ WARN  \] KonvergoWindow.cpp @ 487 - winging it! 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 596 - resize event: QSize(1938, 1127) 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 596 - resize event: QSize(1920, 1080) 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 479 - QWindow visibility set to QWindow::FullScreen 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 257 - Restoring geo: QRect(4110,129 982x551) 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 209 - Load FullScreen geo... 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 294 - setting fullscreen =  false 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 479 - QWindow visibility set to QWindow::Windowed 
2022-06-03 03:02:20 \[ WARN  \] KonvergoWindow.cpp @ 487 - winging it! 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 596 - resize event: QSize(1938, 1127) 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 596 - resize event: QSize(1920, 1080) 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 479 - QWindow visibility set to QWindow::FullScreen 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 257 - Restoring geo: QRect(4110,129 982x551) 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 209 - Load FullScreen geo... 
2022-06-03 03:02:20 \[ INFO  \] JS: Sending web socket message: KeepAlive 
2022-06-03 03:02:20 \[ INFO  \] JS: Received KeepAlive from server. 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 294 - setting fullscreen =  false 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 479 - QWindow visibility set to QWindow::Windowed 
2022-06-03 03:02:20 \[ WARN  \] KonvergoWindow.cpp @ 487 - winging it! 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 596 - resize event: QSize(1938, 1127) 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 596 - resize event: QSize(1920, 1080) 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 479 - QWindow visibility set to QWindow::FullScreen 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 257 - Restoring geo: QRect(4110,129 982x551) 
2022-06-03 03:02:20 \[ DEBUG \] KonvergoWindow.cpp @ 209 - Load FullScreen geo... 
2022-06-03 03:02:21 \[ DEBUG \] KonvergoWindow.cpp @ 294 - setting fullscreen =  false 
2022-06-03 03:02:21 \[ DEBUG \] KonvergoWindow.cpp @ 479 - QWindow visibility set to QWindow::Windowed 
2022-06-03 03:02:21 \[ WARN  \] KonvergoWindow.cpp @ 487 - winging it! 
2022-06-03 03:02:21 \[ DEBUG \] KonvergoWindow.cpp @ 596 - resize event: QSize(1938, 1127) 
2022-06-03 03:02:21 \[ DEBUG \] KonvergoWindow.cpp @ 596 - resize event: QSize(1920, 1080) 
2022-06-03 03:02:21 \[ DEBUG \] KonvergoWindow.cpp @ 479 - QWindow visibility set to QWindow::FullScreen 
2022-06-03 03:02:21 \[ DEBUG \] KonvergoWindow.cpp @ 257 - Restoring geo: QRect(4110,129 982x551) 
2022-06-03 03:02:21 \[ DEBUG \] KonvergoWindow.cpp @ 209 - Load FullScreen geo... 
2022-06-03 03:02:50 \[ INFO  \] JS: Sending web socket message: KeepAlive 
2022-06-03 03:02:50 \[ INFO  \] JS: Received KeepAlive from server. 
2022-06-03 03:03:20 \[ INFO  \] JS: Sending web socket message: KeepAlive 
2022-06-03 03:03:20 \[ INFO  \] JS: Received KeepAlive from server. 
2022-06-03 03:03:50 \[ INFO  \] JS: Sending web socket message: KeepAlive 
2022-06-03 03:03:50 \[ INFO  \] JS: Received KeepAlive from server. 
2022-06-03 03:04:18 \[ DEBUG \] KonvergoWindow.cpp @ 479 - QWindow visibility set to QWindow::Hidden 
2022-06-03 03:04:18 \[ DEBUG \] KonvergoWindow.cpp @ 596 - resize event: QSize(1920, 1080)
```

The only solution is the one that @mekemahmud suggested, which is to edit the `jellyfinmediaplayer.conf` file. The file appears to be located in Local appdata and did not appear on searches (at least for me it didn't), but can be found by going to `%localappdata%/Jellyfinmediaplayer/` or by simply opening `%localappdata%/JellyfinMediaPlayer/jellyfinmediaplayer.conf` in notepad++

**Comment by iwalton3** [comment]
Well there really is just code in here that sets the view to fullscreen on Windows in some conditions. Not sure why I have never had it happen in testing...

```c++
#ifdef Q_OS_WIN32
  if (visibility == QWindow::Windowed)
  {
    QScreen* realScreen = findCurrentScreen();
    if (realScreen && realScreen->geometry() == geometry())
    {
      QLOG_WARN() << "winging it!";
      setScreen(realScreen);
      setVisibility(QWindow::FullScreen);
      return;
    }
  }
#endif
```

**Comment by iwalton3** [comment]
This change will be in this next hotfix. Please let me know if this fixes the issue.

**Comment by DexLuther** [comment]
Seems fixed. Everything I've tried to break it fails to work, and I'm able to pop in and out of fullscreen using any of the shortcuts even having closed the app while in fullscreen.

**Comment by tevenfeng** [comment]
Emm I'm still get the same problem on Windown 11 with Jellyfin Media Player version 1.7.1。

**Comment by tevenfeng** [comment]
> Emm I'm still get the same problem on Windown 11 with Jellyfin Media Player version 1.7.1, even after I uninstalled it and removed all the config files in %APPDATA% and reinstalled it. It always open in fullscreen mode, and alt+enter does nothing, and the titlebar is missing.

I've found out what the problem is. It may be related to the scaling ratio in windows setting. My laptop has a 2k display so I use 150% scaling for display. And jellyfin media player could not handle this situation correctly, making the titlebar out of the display so I misunderstood that it was stuck in fullscreen mode. I reset to 100% scaling, resize the jellyfin window and then change scaling back to 150%, and finallly it worked perfectly.

**Comment by tevenfeng** [comment]
> Emm I'm still get the same problem on Windown 11 with Jellyfin Media Player version 1.7.1, even after I uninstalled it and removed all the config files in %APPDATA% and reinstalled it. It always open in fullscreen mode, and alt+enter does nothing, and the titlebar is missing.

I've found out what the problem is. It may be related to the scaling ratio in windows setting. My laptop has a 2k display so I use 150% scaling for display. And jellyfin media player could not handle this situation correctly, making the titlebar out of the display so I misunderstood that it was stuck in fullscreen mode. I reset to 100% scaling, resize the jellyfin window and then change scaling back to 150%, and finally it worked perfectly.

**Comment by Tychus** [comment]
Can confirm this is actually a scaling issue. Happened to me on windows 11 (100% scaling 2K) changed scaling to 125%  the bar showed up maxed the window then scaled back to 100% all good now. First time i ever get this issue nothing changed just randomly after waking up from sleep.

**Comment by DexLuther** [comment]
> Can confirm this is actually a scaling issue. Happened to me on windows 11 (100% scaling 2K) changed scaling to 125% the bar showed up maxed the window then scaled back to 100% all good now. First time i ever get this issue nothing changed just randomly after waking up from sleep.

Completely different issue. This issue was litterally that the fullscreen check box didn't update the config correctly, which either was always stuck in fullscreen, or wouldn't go into fullscreen. It was fixed a while ago, and this issue should have probably been closed.

**Comment by iwalton3** [comment]
Closing fixed and stale issue.

**Comment by larkwiot** [comment]
Having this issue too, except neither editing nor deleting that configuration file fixes it. Had to uninstall and delete both directories in `AppData\Local` (opened another issue because that should have been done automatically with the uninstallation https://github.com/jellyfin/jellyfin-media-player/issues/506) to get it back in windowed mode.

**Comment by larkwiot** [comment]
> Closing fixed and stale issue.

This issue is not fixed as of 1.9.1

**Comment by Renari** [comment]
Just had this start happening today on 1.11.1, tried a bunch of things in this thread and was unable to resolve the issue. Did an uninstall, deleted all configs and reinstalled and now it lets me exit fullscreen again.

**Comment by Renari** [comment]
@iwalton3 Would you reopen this since it seems this is still occurs.

**Comment by Renari** [comment]
Killing the JMP process while watching something in fullscreen did not reproduce the issue. So I'm in the dark atm since after the uninstall I can't reproduce the issue anymore.

**Comment by greenpean** [comment]
> I have this issue on 1.12.0
> 
> Maybe I gotta clear my cache or something?
> 
> Edit: Clearing cache did the trick!

How to clean the cache?

**Comment by CarterMoody** [comment]
What fixed it for me:
1. Uninstall Jellyfin Media Player
2. Manually remove the Jellyfin Media Player folders from %appdata%/local/
3. Install Jellyfin Media Player

For me, this is really a bug where the title bar doesn't show up. It is as if it is stuck in bordeless fullscreen mode.
This may or may not still be an issue. I dealt with this for a year plus and even upgraded from Windows 10 to 11 but not until now did I remove the appdata folders above.

**Comment by CarterMoody** [comment]
> > I have this issue on 1.12.0
> > Maybe I gotta clear my cache or something?
> > Edit: Clearing cache did the trick!
> 
> How to clean the cache?

at least on Windows, see my reply above about deleting folders

**Comment by andrewrabert** [comment]
There were a few fullscreen related issues fixed in the qt 6 migration - particularly relating to how the app remembers what size/state to restore to when launching again.

Should be fixed, but feel free to reopen if you notice any issues with the latest master.
